File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    c = list(s)
    num_to_alpha = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    
    sb = []

    for cha in c:
        if cha.isdigit():
            answer = answer * 10 + int(cha)
        else:
            sb.append(cha)

            for j in range(10):
                if num_to_alpha[j] == ''.join(sb):
                    answer = answer * 10 + j
                    sb.clear()
                    break

    return answer

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        char[] c = s.toCharArray();
        String[] numToAlpha = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        
        StringBuffer sb = new StringBuffer();

        for (int i = 0; i < c.length; i++) {
            char cha = c[i];
            
            if (Character.isDigit(cha)) {
                answer = answer * 10 + (int) (cha - '0');
            } else {
                sb.append(cha);

                for (int j = 0; j < 10; j++) {
                    if (numToAlpha[j].equals(sb.toString())) {
                        answer = answer * 10 + j;
                        sb.delete(0, sb.length());
                        break;
                    }
                }
            }
        }
        return answer;
    }
}